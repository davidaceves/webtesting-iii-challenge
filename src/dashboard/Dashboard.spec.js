import React from "react";
import * as rtl from "react-testing-library";
import renderer from "react-test-renderer";

import Dashboard from "./Dashboard.js";
import Display from "../display/Display.js";
import Controls from "../controls/Controls.js";

import { fireEvent } from "react-testing-library";

afterEach(rtl.cleanup);

describe("<Dashboard />", () => {
  xit("matches snapshot", () => {
    const tree = renderer.create(<Dashboard />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("defaults", () => {
  xit("defaults to unlocked and open", () => {
    rtl.render(<Dashboard />).getByText(/unlocked/i);
    rtl.render(<Dashboard />).getByText(/open/i);
  });

  it("cannot be closed or opened if it is locked", () => {
    const condition = rtl.render(<Controls locked={true} closed={true} />);
    const button = condition.getByTestId("toggleClosed");

    expect(button.disabled).toBe(true);
  });
});
