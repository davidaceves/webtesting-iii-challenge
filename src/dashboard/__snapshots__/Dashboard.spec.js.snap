// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="toggleLock"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="toggleClosed"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;

exports[`defaults renders controls and display 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="toggleLock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="toggleClosed"
  >
    Close Gate
  </button>
</div>
`;

exports[`defaults renders controls and display 2`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
